### PR TITLE
Mh io batch

### PR DESCRIPTION
**Pull request by DStrelak**

### CODE REVIEW

File: core/rwMRC.cpp [hunk 1]
```diff
@@ -183,16 +185,22 @@ int ImageBase::readMRC(size_t select_img, bool isStack)
...
+        if ( start_img > _nDim )
+            REPORT_ERROR(ERR_INDEX_OUTOFBOUNDS, formatString("readMRC: %s Image number %lu exceeds stack size %lu", this->filename.c_str(), start_img, _nDim));
 
-        if (select_img > ALL_IMAGES)
-            _nDim = 1;
     }
     else // If the reading is not like a stack, then the select_img is not taken into account and must be selected the only image
     {
-        select_img = ALL_IMAGES;
+        start_img = 1;
```
Comment (DStrelak) [inline core/rwMRC.cpp:201]: are you sure that it's 1?

File: core/rwMRC.cpp [hunk 2]
```diff
@@ -604,25 +627,31 @@ int ImageBase::writeMRC(size_t select_img, bool isStack, int mode, const String
...
-    {
-        // If to also write the image data or jump its size
-        if (dataMode >= DATA)
+    if (checkMmapT(wDType) && !mmapOnWrite && dataMode >= DATA) {
+        writeData(fimg, 0, wDType, datasize_n * imgEnd, castMode);
+    } else {
+        for ( size_t i = 0; i < imgEnd; i++ )
         {
-            if (mmapOnWrite && Ndim == 1) // Can map one image at a time only
+            // If to also write the image data or jump its size
```
Comment (DStrelak) [inline core/rwMRC.cpp:635]: ?

File: core/rwMRC.cpp (hunk 1 above)
Comment (horeso) [reply core/rwMRC.cpp:201]: This part of if branch is visited only when `isStack==false`. The line  `_nDim=1;` is from devel, therefore the file contains only one image if isStack is false.

File: core/rwMRC.cpp (hunk 2 above)
Comment (horeso) [reply core/rwMRC.cpp:635]: This is how the code looks now: 
image


This is original version:  
image

The only change that happened is highlighted by blue color. Github treats strangely the whole part of code from the screenshot.

**Pull request merged**